src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -632,4 +636,84 @@ lemma units_mod_principal_units_equiv_residue_field_units_comp_quotient_group_mk
 
 end principal_unit_group
 
+/-! ### Pointwise actions
+
+This transfers the action from `subring.pointwise_mul_action`, noting that it only applies when
+the action is by a group. Notably this provides an instances when `G` is `K ≃+* K`.
+
+These instances are in the `pointwise` locale.
+
+The lemmas in this section are copied from `ring_theory/subring/pointwise.lean`; try to keep these
+in sync.
+-/
+section pointwise_actions
+open_locale pointwise
+
+variables {G : Type*} [group G] [mul_semiring_action G K]
+
+/-- The action on a valuation subring corresponding to applying the action to every element.
+
+This is available as an instance in the `pointwise` locale. -/
+def pointwise_has_smul : has_smul G (valuation_subring K) :=
+{ smul := λ g S,
+  { mem_or_inv_mem' := λ x, (mem_or_inv_mem S (g⁻¹ • x)).imp
+      (subring.mem_pointwise_smul_iff_inv_smul_mem.mpr)
+      (λ h, subring.mem_pointwise_smul_iff_inv_smul_mem.mpr $ by rwa smul_inv''),
+    .. g • S.to_subring } }
```

Comment:
It's annoying that this proof is non-slick. For the comap definition it's just
```
def valuation_subring.comap  (A : valuation_subring L) (e  : K →+* L) :
  valuation_subring K :=
{ mem_or_inv_mem' := λ k, by simp [valuation_subring.mem_or_inv_mem],
  ..(A.to_subring.comap e) }
```

Reply:
Everything else looks very nice though.

Reply:
I didn't attempt to try to prove it that way; the advantage of the term mode proof is that it makes it clear _why_ `group G` is needed, whereas an unsqueezed simp call would not.

It's also possible that writing `map` directly is easier than working with the `smul` API, but I figured I'd leave defining `valuation_ring.map` to whoever ends up needing it